scripts/addRif.py

```diff
@@ -0,0 +1,126 @@
+# Copyright (C) University of Tennessee Health Science Center, Memphis, TN.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Affero General Public License
+# as published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# This program is available from Source Forge: at GeneNetwork Project
+# (sourceforge.net/projects/genenetwork/).
+#
+# Contact Drs. Robert W. Williams and Xiaodong Zhou (2010)
+# at rwilliams@uthsc.edu and xzhou15@uthsc.edu
+#
+#
+#
+# This module is used by GeneNetwork project (www.genenetwork.org)
+#
+# Created by GeneNetwork Core Team 2010/08/10
+#
+# Last updated by Lei Yan 2011/02/08
+
+# created by Lei Yan 02/08/2011
+
+import string
+import MySQLdb
+import time
+import os
+import sys
+
+path1 = os.path.abspath(os.path.dirname(__file__))
+path2 = path1 + "/.."
+path3 = path1 + "/../../tmp"
+sys.path.insert(0, path2)
```

Comment:
Adding a path to the system paths like this, I find, should be extremely rare, and if possible, avoided. I usually make the script directory a package and use then would do something like `…/python3 -m scripts.addRif [ARGS&OPTS  …]` at the root of the repository - avoids having to mess with paths like this.

With the paths hard-coded like this, if you were to, for some reason, need to move this script to say a sub-package, or elsewhere in the repo, the path system you've set up above will very likely break. On the other hand, the `…/python3 -m <package>[.<package>]*.<module>` will just pick up the paths automatically.